src/lib.rs

```diff
@@ -742,12 +742,15 @@ impl Manifest {
             .map_err(|_e| ManifestError::MissingManifest(manifest_path_buf))?;
         let mut manifest: Self = toml::from_str(contents.as_str())?;
 
-        if manifest.package.readme.is_none() {
-            manifest.package.readme = locate_file(path, README_PATHS);
-        }
+        if let Some(mut package) = manifest.package.clone() {
+            if package.readme.is_none() {
+                package.readme = locate_file(path, README_PATHS);
+            }
 
-        if manifest.package.license_file.is_none() {
-            manifest.package.license_file = locate_file(path, LICENSE_PATHS);
+            if package.license_file.is_none() {
+                package.license_file = locate_file(path, LICENSE_PATHS);
+            }
+            manifest.package = Some(package);
```

Comment:
nit: I would call `manifest.package.as_mut` to modify the package in-place to prevent cloning and then rewriting the original `package`. But it doesn't block from merging.

Reply:
oooh nice! thanks for the tip, I didn't know about `as_mut`.